packages/loader/container-loader/src/loader.ts

```diff
@@ -455,6 +460,35 @@ export class Loader implements IHostLoader {
         return { container, parsed };
     }
 
+    private async resolveWithDomainChangeHandling(
+        request: IRequest,
+        pendingLocalState?: IPendingLocalState,
+    ): Promise<{ container: Container; parsed: IParsedUrl }> {
+        let req: IRequest = request;
+        let success = false;
+        let response;
+        do {
+            const resolvedUrl = await this.resolveUrl(req);
+            try {
+                response = await this.getContainer(resolvedUrl, req, pendingLocalState);
+                success = true;
+            } catch (error: any) {
+                if (isDomainMoveError(error)) {
+                    const absoluteUrl = await this.services.urlResolver.getAbsoluteUrl(
```

Comment:
I think direction is reasonable, but I find it odd that you did not change signature of getAbsoluteUrl() on interface but use it here.
I'd say I'd rather prefer a new method, not jam it into existing method.